applications/tasks/src/modules/kt_rates.erl

```diff
@@ -0,0 +1,405 @@
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".output_header">>, ?MODULE, 'output_header'),
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".direction">>, ?MODULE, 'direction'),
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".cleanup">>, ?MODULE, 'cleanup'),
+    tasks_bindings:bind_actions(<<"tasks."?CATEGORY>>, ?MODULE, ?ACTIONS).
+
+-spec output_header(ne_binary()) -> kz_csv:row().
+output_header(<<"export">>) -> [?DOC_FIELDS].
+
+-spec help(kz_json:object()) -> kz_json:object().
+help(JObj) -> help(JObj, <<?CATEGORY>>).
+
+-spec help(kz_json:object(), ne_binary()) -> kz_json:object().
+help(JObj, <<?CATEGORY>>=Category) ->
+    lists:foldl(fun(Action, J) -> help(J, Category, Action) end, JObj, ?ACTIONS).
+
+-spec help(kz_json:object(), ne_binary(), ne_binary()) -> kz_json:object().
+help(JObj, <<?CATEGORY>>=Category, Action) ->
+    kz_json:set_value([Category, Action], kz_json:from_list(action(Action)), JObj).
+
+-spec action(ne_binary()) -> kz_proplist().
+action(<<"export">>) ->
+    [{<<"description">>, <<"Export ratedeck">>}
+    ,{<<"doc">>, <<"Export rates from \"ratedeck\" DB">>}
+    ];
+
+action(<<"import">>) ->
+    %% prefix & cost is mandatory fields
+    {Mandatory, Optional} = lists:split(2, [?DOC_FIELDS]),
+    [{<<"description">>, <<"Bulk-import rates">>}
+    ,{<<"doc">>, <<"Creates rates from file">>}
+    ,{<<"expected_content">>, <<"text/csv">>}
+    ,{<<"mandatory">>, Mandatory}
+    ,{<<"optional">>, Optional}
+    ];
+
+action(<<"delete">>) ->
+    %% prefix is mandatory field
+    {Mandatory, Optional} = lists:split(1, [?DOC_FIELDS]),
+    [{<<"description">>, <<"Bulk-remove rates">>}
+    ,{<<"doc">>, <<"Delete rates from file">>}
+    ,{<<"expected_content">>, <<"text/csv">>}
+    ,{<<"mandatory">>, Mandatory}
+    ,{<<"optional">>, Optional}
+    ].
+
+%%% Verifiers
+
+-spec direction(ne_binary()) -> boolean().
+direction(<<"inbound">>) -> 'true';
+direction(<<"outbound">>) -> 'true';
+direction(_) -> 'false'.
+
+%%% Appliers
+
+-spec export(kz_proplist(), task_iterator()) -> task_iterator().
+export(Props, 'init') ->
+    case is_allowed(Props) of
+        'true' -> State = [{'db', get_ratedeck_db(Props)}
+                          ,{'options', [{'limit', ?BULK_LIMIT + 1}
+                                       ,'include_docs'
+                                       ]
+                           }
+                          ],
+                  export(Props, State);
+        'false' ->
+            lager:warning("rates exporting is forbidden for account ~s, auth account ~s"
+                         ,[props:get_value('account_id', Props)
+                          ,props:get_value('auth_account_id', Props)
+                          ]
+                         ),
+            {<<"task execution is forbidden">>, 'stop'}
+    end;
+export(_Props, 'stop') -> 'stop';
+export(_Props, State) ->
+    Db = props:get_value('db', State),
+    Options = props:get_value('options', State),
```

Comment:
I'm rewriting some things in tasks for a client & Props will now be a map. The variable amount of arguments an input task takes will also be merged into one map (one map per csv row).